source/common/http/match_delegate/config.cc

```diff
@@ -14,7 +14,7 @@ namespace Common {
 namespace Http {
 namespace MatchDelegate {
 
-namespace {
+namespace Factory {
```

Comment:
not sure whether this name makes sense as the class within it has a Factory suffix anyway but I do not have a strong opinion in here and I assume that this change was needed for some reason.

Reply:
I added this namespace name because I don't think it's possible to refer to anonymous namespaces externally, which is needed to call the `forceRegister...` function. Open to better names.

Reply:
I thought that for unnamed namespaces you could just omit them when referring to members declared as part of them. As in `Http::MatchDelegate::WHATEVER_YOU_WANT_TO_ACCESS` as opposed to `Http::MatchDelegate::Factory::WHATEVER_YOU_WANT_TO_ACCESS`

Reply:
Here's the compilation failure I get when trying to access things this way:

```
external/envoy/source/common/http/match_delegate/config.h:109:1: error: function 'Envoy::Common::Http::MatchDelegate::(anonymous namespace)::forceRegisterSkipActionFactory' has internal linkage but is not defined [-Werror,-Wundefined-internal]
DECLARE_FACTORY(SkipActionFactory);
^
external/envoy/envoy/registry/registry.h:660:61: note: expanded from macro 'DECLARE_FACTORY'
#define DECLARE_FACTORY(FACTORY) ABSL_ATTRIBUTE_UNUSED void forceRegister##FACTORY()
                                                            ^
<scratch space>:25:1: note: expanded from here
forceRegisterSkipActionFactory
^
external/envoy_build_config/extension_registry.cc:68:32: note: used here
  Common::Http::MatchDelegate::forceRegisterSkipActionFactory();
                               ^
1 error generated.
```

Reply:
Got it, thank you for the explanation & error message.